File: frontend/types/audit.ts
export interface AuditLog {
    id: number;
    user_id: number | null;
    user_name: string;
    username: string;
    action: string;
    resource_type: string;
    resource_id: string | null;
    timestamp: string;
    metadata: Record<string, any> | null;
}

export interface AuditLogListResponse {
    items: AuditLog[];
    total: number;
    page: number;
    size: number;
}

File: backend/app/schemas/audit.py
from pydantic import BaseModel, Field
from typing import Any, Dict

class AuditLogResponse(BaseModel):
    id: int
    user_id: int | None
    user_name: str
    username: str
    action: str
    resource_type: str
    resource_id: str | None
    timestamp: str
    metadata: Dict[str, Any] | None = None

class AuditLogListResponse(BaseModel):
    items: list[AuditLogResponse]
    total: int
    page: int
    size: int
